libs/core/src/main/java/org/opensearch/Version.java

```diff
@@ -160,15 +234,11 @@ private static Version fromIdSlow(int id) {
         } else {
             luceneVersion = versions.get(index).luceneVersion;
         }
-        return id < MASK ? new LegacyESVersion(id, luceneVersion) : new Version(id ^ MASK, luceneVersion);
-    }
-
-    public static int computeLegacyID(int major, int minor, int revision, int build) {
-        return major * 1000000 + minor * 10000 + revision * 100 + build;
+        return new Version(id ^ MASK, luceneVersion);
     }
 
     public static int computeID(int major, int minor, int revision, int build) {
-        return computeLegacyID(major, minor, revision, build) ^ MASK;
+        return (major * 1000000 + minor * 10000 + revision * 100 + build) ^ MASK;
```

Comment:
nit: Let's store the multipliers as constants - to avoid counting the zeros whenever/wherever we are using them.
Also, nice to use underscores (`1_000_000`) in these numeric constants as well.

Reply:
It doesn't make sense to keep track of them in terms of thousands. 

Maybe as powers of 100? The whole point is that each part gets 2 digits (which is why the semver ranges that assumed a version like `999.999.999` was valid were always broken). 

I guess I can do something like:

```
private static final int VERSION_SHIFT = 100;
private static final int REVISION_SHIFT = VERSION_SHIFT;
private static final int MINOR_SHIFT = VERSION_SHIFT * VERSION_SHIFT;
private static final int MAJOR_SHIFT = VERSION_SHIFT * VERSION_SHIFT * VERSION_SHIFT;
```